### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by ndv20**
[DEMO LINK](https://dmytronykorovych.github.io/react_movies-list-add-form/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,16 +1,26 @@
+import { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
+
+const moviesList: Movie[] = moviesFromServer.map(movie => ({ ...movie }));
```
Comment (VitaliyBodnar) [inline src/App.tsx:8]:
```suggestion
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
 
 export const App = () => {
+  const [movies, setMovies] = useState(moviesList);
```
Comment (VitaliyBodnar) [inline src/App.tsx:11]: redundant variable 
```suggestion
  const [movies, setMovies] = useState<Movie[]>(moviesFromServer);
```
Missed type for useState

File: src/components/NewMovie/NewMovie.tsx [hunk 3]
```diff
@@ -1,45 +1,92 @@
...
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={(value: string) => setTitle(value)}
```
Comment (VitaliyBodnar) [inline src/components/NewMovie/NewMovie.tsx:57]: Can be less
```suggestion
        onChange={setTitle}
```
Check other places

**Review by VitaliyBodnar — CHANGES_REQUESTED** [review]

**Review by volodymyr-soltys97 — APPROVED** [review]
Well done 👍

**Pull request closed without merging**